public/app/features/dashboard/components/DashboardSettings/TimePickerSettings.tsx

```diff
@@ -76,6 +114,26 @@ export class TimePickerSettings extends PureComponent<Props, State> {
             defaultValue={this.props.nowDelay}
           />
         </Field>
+        <Field
+           label="Max time range"
+           description="Prevent users from choosing a time range larger than a specified time interval. The supported units are y(years), M(months), w(weeks), d(days), h(hours), m(minutes), s(seconds)."
+        >
+          <Input
+            invalid={!this.state.isMaxTimeRangeValid}
+            onChange={this.onMaxTimeRangeChange}
+            defaultValue={this.props.maxTimeRange}
+          />
+        </Field>
+        <Field
+           label="Oldest 'From' now -"
```

Comment:
If we use this (label="Oldest 'from' time") then this will be confusing to the user. It seems that the user may think that he/she has to input now-<time>.